### PR TITLE
Precedence tests

### PR DESCRIPTION
**Pull request by mmatera**
This includes the tests proposed in https://github.com/Mathics3/mathics-core/pull/1192 where it should belong.

### CODE REVIEW

**Comment by mmatera** [comment]
Whether the tests could be implemented at a lower level, just looking into the tables, I want to include later some tests checking how certain expressions are parsed and formatted when the precedence values are taken into account, so I need Mathics-core for that.

**Comment by mmatera** [comment]
No, the problem was that the github action workflow uses the [operator-info-from-JSON](https://github.com/Mathics3/mathics-core/tree/operator-info-from-JSON) branch of Mathics-core. Once I updated the branch, the test passed.

File: test/test_mathics_precedence.py [hunk 1]
```diff
@@ -0,0 +1,606 @@
...
+    "Repeated",
+    "RepeatedNull",
+    "SuchThat",
+    "DoubleLeftTee",
+    "DoubleRightTee",
+    "DownTee",
+    "LeftTee",
+    "Perpendicular",  # 190
+    "RightTee",  # 190
+    # In WMA, `RoundImplies` has a
```
Comment (mmatera) [inline test/test_mathics_precedence.py:308]: We probably need to adjust the  precedence values in the tables for `UpTee` and `RoundImplies`

File: test/test_mathics_precedence.py [hunk 2]
```diff
@@ -0,0 +1,606 @@
...
+    "RightTriangleEqual",
+    "Succeeds",
+    "SucceedsEqual",
+    "SucceedsSlantEqual",
+    "SucceedsTilde",
+    "Tilde",
+    "TildeEqual",
+    "TildeFullEqual",
+    "TildeTilde",
+    # In Mathics, the precedence of these operators is quite low.
```
Comment (mmatera) [inline test/test_mathics_precedence.py:458]: Is there a reason to chose these lower values for the precedence of these operators?

File: test/test_mathics_precedence.py [hunk 3]
```diff
@@ -0,0 +1,606 @@
...
+    "Wedge",
+    "Diamond",
+    "Backslash",
+    "Divide",
+    "Minus",
+    "Dot",
+    "CircleDot",
+    "SmallCircle",
+    "Square",  # 540
+    "Del",  # In WMA, has the same precedence as DifferentialD and CapitalDifferentialD
```
Comment (mmatera) [inline test/test_mathics_precedence.py:490]: I do not see the need of using a different value of precedence for these operators.

File: test/test_mathics_precedence.py [hunk 4]
```diff
@@ -0,0 +1,606 @@
...
+    #    "UndirectedEdge", # Mathics 120, WMA 295
+    "SquareUnion",
+    "UnionPlus",
+    "Span",
+    "SquareIntersection",
+    "MinusPlus",
+    "PlusMinus",
+    "Plus",
+    "Subtract",  #  310
+    #    "Integrete", # In Mathics, this has the default precedence. In WMA, 325
```
Comment (mmatera) [inline test/test_mathics_precedence.py:469]: At some point, we need to look over again the way how the information for `Integral` and `Integrate` operators are stored in the tables..

**Comment by rocky** [comment]
> No, the problem was that the github action workflow uses the [operator-info-from-JSON](https://github.com/Mathics3/mathics-core/tree/operator-info-from-JSON) branch of Mathics-core. Once I updated the branch, the test passed.

Ah, that was a step  I forgot to do after the  branch got merged.

File: test/test_mathics_precedence.py (hunk 3 above)
Comment (rocky) [reply test/test_mathics_precedence.py:490]: Ok. Put in a _isolated_ PR for this change along with changing the test to work from master. If the tests succeed, we can merge that in and delete operators_from_JSON branch in mathics-core.

File: test/test_mathics_precedence.py (hunk 2 above)
Comment (rocky) [reply test/test_mathics_precedence.py:458]: > Is there a reason to choose these lower values for the precedence of these operators?

See https://github.com/Mathics3/mathics-scanner/pull/99#discussion_r1863620711

File: test/test_mathics_precedence.py (hunk 3 above)
Comment (mmatera) [reply test/test_mathics_precedence.py:490]: Yep, this is just about the tests. Here I just wanted to leave the comments.

File: test/test_mathics_precedence.py (hunk 1 above)
Comment (rocky) [reply test/test_mathics_precedence.py:308]: See https://github.com/Mathics3/mathics-scanner/pull/99#discussion_r1863620711

**Comment by mmatera** [comment]
The tests are already merged

**Pull request closed without merging**